### PR TITLE
Issue78: Implement match_orders using order equivalence

### PR DESCRIPTION
**Pull request by DanilYachmenev**
Task https://github.com/sorrentum/sorrentum/issues/120

Follow-up from https://github.com/sorrentum/sorrentum/issues/78#issuecomment-1500220065

Order equivalence is introduced
DRAFT

nits from other PRs are not moved yet

### CODE REVIEW

File: defi/dao_cross/dao_cross.py [hunk 1]
```diff
@@ -93,7 +105,51 @@ def match_orders(
...
+        _LOG.warning("Sell orders remain unmatched: %s", sell_heap)
+    return transfers_df
+
+
+def get_equivalent_order(
+    order: ddacrord.Order,
+    clearing_price: float,
+) -> order: ddacrord.Order:
+    """
+    Get equivalent DaoCross order.
```
Comment (tamriq) [inline defi/dao_cross/dao_cross.py:121]: Let's add en example to the docstring. E.g. ("buy", "BTC", 1, "ETH", 5) - > ("sell", "ETH", 5, "BTC", 1)

File: defi/dao_cross/dao_cross.py [hunk 2]
```diff
@@ -21,12 +21,16 @@
...
+    base_token: str,
+    quote_token: str,
 ) -> pd.DataFrame:
     """
-    Implement DaoCross orders matching.
+    Implement DaoCross orders matching for token swaps.
 
     :param orders: orders to match
     :param clearing_price: clearing price
+    :param base_token: name of the base token for swaps
```
Comment (tamriq) [inline defi/dao_cross/dao_cross.py:32]: "name of the base token for swaps, which determines the quantity"

File: defi/dao_cross/dao_cross.py [hunk 3]
```diff
@@ -21,12 +21,16 @@
...
+    :param quote_token: name of the quote token for swaps
```
Comment (tamriq) [inline defi/dao_cross/dao_cross.py:33]: "name of the base token for swaps, which determines the price"

File: defi/dao_cross/dao_cross.py (hunk 1 above)
Comment (tamriq) [inline defi/dao_cross/dao_cross.py:121]: Let's add en example to the docstring. E.g. ("buy", "BTC", 1, "ETH", 5) - > ("sell", "ETH", 5, "BTC", 1)

**Review by gpsaggese — COMMENTED** [review]
LGTM

**Comment by DanilYachmenev** [comment]
Reimplemented in https://github.com/sorrentum/sorrentum/pull/137 since this one broke due to branch-to-branch merges

**Pull request closed without merging**